cilium/CFP-25694-move-cilium-cli-to-cilium-repo.md

```diff
@@ -0,0 +1,167 @@
+- Release [tags](https://github.com/cilium/cilium-cli/tags) and [artifacts](https://github.com/cilium/cilium-cli/releases)
+
+## Impacts / Key Questions
+
+### Impact: Backporting to stable branches
+
+Starting in `v1.15`, you need to backport bug fixes related to
+`cilium connectivity test` command from `main` branch to affected stable
+branches since these stable branches use their own copy of `cli/` to run
+`cilium connectivity test`. Apply the same [backporting process] as the rest of
+the Cilium codebase.
+
+### Impact: Module name change
+
+This proposal impacts users who depend on [cilium/cilium-cli] as a library.
+
+- The module path needs to be updated from `github.com/cilium/cilium-cli` to
+  `github.com/cilium/cilium/cli`.
+- Instead of depending on a version of `github.com/cilium/cilium-cli` (`v0.14.5`
+  for example), you need to inspect `go.mod` of a specific `github.com/cilium/cilium-cli`
+  version, and transitively find the `github.com/cilium/cilium/cli` version to
+  depend on.
+- Alternatively, depend on a particular `github.com/cilium/cilium` version without
+  inspecting `go.mod` file in `github.com/cilium/cilium-cli`.
+
+If you must retain the Go module name, you need to copy the `cilium-cli` code
+back to [cilium/cilium-cli] repo and modify all the import paths back to
+`github.com/cilium/cilium-cli` for each release..
+See https://github.com/cilium/design-cfps/pull/9#discussion_r1211996796 for details.
+
+### Key Question: How do you run CI for CLI? More specifically, how do you ensure that changes to Cilium CLI remain compatible with older released versions of Cilium?
+
+Here is the list of GitHub workflows that currently run for each pull requests in [cilium/cilium-cli]
+using the latest version of Cilium:
+
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/aks-byocni.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/eks-tunnel.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/eks.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/externalworkloads.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/gke.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/kind.yaml
+- https://github.com/cilium/cilium-cli/blob/main/.github/workflows/multicluster.yaml
+
+[cilium/cilium] has a similar coverage using CI images instead of released images,
+so it's probably an overkill to port all of these workflows to [cilium/cilium]. My
+proposal is to move https://github.com/cilium/cilium-cli/blob/main/.github/workflows/kind.yaml
+to [cilium/cilium], and run this workflow for pull requests that modify files
+under `cli/` directory. This ensures that any changes under `cli/` directory get
+tested against a released version of Cilium.
+
+## Alternative approach: Extend connectivity test from cilium/cilium repo
```

Comment:
looking at https://github.com/cilium/cilium-cli/pull/1683 as an example of extending `cilium connectivity test`, i'm not sure how well this approach works in practice. maybe the hook interface needs to be extended to support stuff like:

- add new flags to `cilium connectivity test` command
- extend the setup logic